src/api/firebase.js

```diff
@@ -71,8 +73,44 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, docId, itemData) {
 	const docRef = doc(db, listId, docId);
+	const { isChecked, totalPurchases, dateLastPurchased, dateNextPurchased } =
+		itemData;
+	let data;
 
-	await updateDoc(docRef, itemData);
+	const currentDate = new Date();
+	const oldDateLastPurchased = dateLastPurchased
+		? dateLastPurchased.toDate()
+		: currentDate;
+	const oldDateNextPurchased = dateNextPurchased.toDate();
+	const previousEstimate = getDaysBetweenDates(
+		oldDateLastPurchased,
+		oldDateNextPurchased,
+	);
+
+	const daysSinceLastTransaction = getDaysBetweenDates(
+		oldDateLastPurchased,
+		currentDate,
+	);
+
+	const estimatedNextPurchaseInDays = calculateEstimate(
+		previousEstimate,
+		daysSinceLastTransaction,
+		totalPurchases,
+	);
+
+	const newDateNextPurchased = addDaysToDate(
+		currentDate,
+		estimatedNextPurchaseInDays,
+	);
+
+	data = {
+		isChecked,
+		totalPurchases,
+		dateLastPurchased: Timestamp.fromDate(currentDate),
```

Comment:
This value should always be updated to the current date, since that's when we're buying the item.